packages/victory-core/src/victory-state/victory-provider.tsx

```diff
@@ -14,27 +14,42 @@ export interface ContextType {
   data: FormattedDatum[];
   scale: ScaleType;
   domain: DomainType;
-  setChildProps: (id: string, props: VictoryProviderProps) => void;
+  updateChildProps: (id: symbol, props: VictoryProviderProps | null) => void;
 }
 
 const VictoryContext = createContext<ContextType | null>(null);
 
-export function VictoryProvider({
-  children,
-  ...initialProps
-}: VictoryProviderProps) {
-  // We need to store the props in state so they can be overwritten by child components
-  const [props, setProps] = React.useState(initialProps);
+type CollectedProps = { [id: symbol]: VictoryCalculatedStateProps };
 
-  const setChildProps = React.useCallback(
-    (id: string, newProps: VictoryCalculatedStateProps) => {
-      setProps((prevProps) => {
-        return { ...prevProps, ...newProps };
+function useNormalizedProps(): ContextType {
+  const [collectedProps, setCollectedProps] = React.useState<CollectedProps>(
+    {},
+  );
+
+  const updateChildProps = React.useCallback(
+    (id: symbol, newProps: VictoryCalculatedStateProps | null) => {
+      setCollectedProps((prev) => {
+        const result = { ...prev };
+        if (newProps === null) {
+          delete result[id];
+        } else {
+          result[id] = newProps;
+        }
+        return result;
       });
     },
     [],
   );
 
+  // TEMP: combine all props into a single result:
+  // TODO: instead, we should intelligently aggregate all these props
```

Comment:
Yep, this is essentially the approach I was thinking about. For some props like `data` we will need to return the specific data for each child, but for something like `domain` or `scale` we will need to get the total values for all the aggregated data.